### PR TITLE
Kotlin migration: SimpleMessageDialog

### PR DESCRIPTION
**Pull request by Arnold2381**
Kotlin migration

### CODE REVIEW

**Review by mikehardy — APPROVED** [review]
LGTM - I'm not sure how the `"" != title` works for Kotlin if title is null, followed by the `title!!` non-null coercion, does kotlin need a null check or is it bulit in there? I guess the previous java implementation would fail in the same way so this is equivalent behavior either way

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/SimpleMessageDialog.kt [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        val title = requireArguments().getString("title")
+        return if ("" != title) {
+            title!!
+        } else {
+            AnkiDroidApp.getAppResources().getString(R.string.app_name)
+        }
+    }
+
+    override fun getNotificationMessage(): String {
+        return requireArguments().getString("message")!!
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/SimpleMessageDialog.kt:58]: This was previously `String?`

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/SimpleMessageDialog.kt [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                    ?.dismissSimpleMessageDialog(
+                        requireArguments().getBoolean(
+                            "reload"
+                        )
+                    )
+            }
+            .show()
+    }
+
+    override fun getNotificationTitle(): String {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/SimpleMessageDialog.kt:48]: This was previously `String?`

**Review by david-allison-1 — CHANGES_REQUESTED** [review]

**Comment by david-allison-1** [comment]
> LGTM - I'm not sure how the `"" != title` works for Kotlin if title is null, followed by the `title!!` non-null coercion, does kotlin need a null check or is it bulit in there? I guess the previous java implementation would fail in the same way so this is equivalent behavior either way

Java would return null, which would work. Kotlin fails here

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/SimpleMessageDialog.kt [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                        )
+                    )
+            }
+            .show()
+    }
+
+    override fun getNotificationTitle(): String? {
+        val title = requireArguments().getString("title")
+        return if ("" != title) {
+            title!!
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/SimpleMessageDialog.kt:51]: don't `!!` here

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/SimpleMessageDialog.kt (hunk 1 above)
Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/anki/dialogs/SimpleMessageDialog.kt:58]: This also needs the `!!` removed

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
One more round

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/SimpleMessageDialog.kt [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    interface SimpleMessageDialogListener {
+        fun dismissSimpleMessageDialog(reload: Boolean)
+    }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): MaterialDialog {
+        // FIXME this should be super.onCreateDialog(Bundle), no?
+        super.onCreate(savedInstanceState)
+        return MaterialDialog.Builder(requireActivity())
+            .title(notificationTitle!!)
+            .content(notificationMessage!!)
```
Comment (Arnold2381) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/SimpleMessageDialog.kt:35]: @david-allison-1 should i follow up with a PR for titlenullable and content nullable here too?

Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/anki/dialogs/SimpleMessageDialog.kt:35]: These are buggy and need to be fixed in this PR

Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/anki/dialogs/SimpleMessageDialog.kt:35]: the PR with the fluent-nullability has landed

**Comment by Arnold2381** [comment]
@david-allison-1 i think this is ready.

**Review by david-allison-1 — APPROVED** [review]
LGTM

**Pull request merged**